ipc/cli/src/commands/wallet/balances.rs

```diff
@@ -21,26 +22,66 @@ impl CommandLineHandler for WalletBalances {
     async fn handle(global: &GlobalArguments, arguments: &Self::Arguments) -> anyhow::Result<()> {
         log::debug!("list wallets with args: {:?}", arguments);
 
-        let url = get_ipc_agent_url(&arguments.ipc_agent_url, global)?;
-        let json_rpc_client = JsonRpcClientImpl::new(url, None);
+        let provider = get_ipc_provider(global)?;
 
-        let params = WalletBalancesParams {
-            subnet: arguments.subnet.clone(),
-        };
+        let wallet_type = WalletType::from_str(&arguments.wallet_type)?;
+        let subnet = Arc::new(SubnetID::from_str(&arguments.subnet)?);
```

Comment:
We might be able to remove the `Arc` and use reference and we dont have to clone.

Reply:
fixed